### ISSUE TITLE
Pure ems bundle prevents node-fetch usage in libraries/frameworks

### ISSUE THREAD

**Issue by darkbasic** [node-fetch/node-fetch#1791]
Even if I think that the developer experience (especially with Typescript) it's not quite there yet I would have no problem switching my own personal project to ESM. Unfortunately as a library developer I cannot force my decisions down everyone else's throat so I must keep my libraries compatible with both esm and commonjs (that is especially true due to the still suboptimal esm experience with Typescript): that's why I decided to build my libraries twice and bundle both cjs and esm.
node-fetch 3 prevents library developers from targeting commonjs because we cannot import pure esm modules if one of our build targets is commonjs.
May I ask what's preventing you from shipping both cjs and esm? For the moment that's not really a problem because you keep backporting fixes into v2, but since for the foreseable future I cannot see me dropping the commonjs target I also want to plan for the long term.

Thanks

**Comment by jimmywarting** [comment]
You can use async import to load esm modules in cjs
https://github.com/node-fetch/node-fetch/issues/1279

**Comment by darkbasic** [comment]
I closed the issue because I decided that it wasn't going to achieve anything at the end of the day and I didn't want to be controversial for no reason, but since you've answered I'll tell you my view.

I guess you don't provide cjs because of dual package hazard concerns. While that is true you also suggest to use async imports in a library which is going to output commonjs code. Even if we don't consider the fact that this is only viable in async contexts, that same hazard exists for esm-only packages that get imported in commonjs projects via async imports so your suggestion goes against the reason why you decided to bundle pure esm in the first place and my library could bite other users because of that!
Also in my case the [whole graphql ecosystem](https://unpkg.com/graphql@16.8.1/index.mjs) bundles dual packages and thus using `instanceof` is already a big hazard (let's not talk about `yarn link` which creates a similar issue).
The developer experience for esm modules on Typescript is still suboptimal because importing commonjs modules with default exports is a mess and the js extension in the imports is counterintuitive to say the least.
I personally think that it's still too early to target pure esm in a library but at the same time you cannot ignore deno/bun existence so dual package is the lesser evil. Just my two cents.
